topic-operator/src/main/java/io/strimzi/operator/topic/BatchingTopicController.java

```diff
@@ -820,13 +820,15 @@ private void warnTooLargeMinIsr(List<ReconcilableTopic> reconcilableTopics) {
         for (ReconcilableTopic reconcilableTopic : reconcilableTopics) {
             var topicConfig = reconcilableTopic.kt().getSpec().getConfig();
             if (topicConfig != null) {
-                Integer topicMinIsr = (Integer) topicConfig.get(KafkaHandler.MIN_INSYNC_REPLICAS);
-                var minIsr = topicMinIsr != null ? topicMinIsr : clusterMinIsr.map(Integer::parseInt).orElse(1);
+                var topicMinIsr = topicConfig.get(KafkaHandler.MIN_INSYNC_REPLICAS);
+                var configuredMinIsr = topicMinIsr != null 
+                    ? Integer.parseInt(TopicOperatorUtil.configValueAsString(KafkaHandler.MIN_INSYNC_REPLICAS, topicMinIsr))
+                    : clusterMinIsr.map(Integer::parseInt).orElse(KafkaHandler.DEFAULT_MIN_ISR);
```

Comment:
Is there some possibility that clusterMinIsr.map(Integer::parseInt) would be null? Because if so it would throw NPE which is not handled.

Reply:
clusterMinIsr is optional, so it can be empty, not null

Reply:
sorry to say that because we had this discussion in the past but using `var` doesn't help at all to understand that `clusterMinIsr` is `Optional` unless I checkout you branch and open it with an IDE or I navigate on GitHub to understand where it's set and by which method.

Reply:
> clusterMinIsr is optional, so it can be empty, not null

Okay, thanks.